In [5]:
import pandas as pd
import json
import requests

In [8]:
r = requests.get('http://localhost:5000/v1/stats/data/full')
df = pd.read_json(json.dumps(r.json()))

## Exploring the data

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103658 entries, 0 to 103657
Data columns (total 18 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   age_6              19184 non-null   float64
 1   cma                103658 non-null  int64  
 2   cma_name           103658 non-null  object 
 3   cowmain            67062 non-null   float64
 4   cowmain_name       103658 non-null  object 
 5   educ               103658 non-null  int64  
 6   full_naics_21      103658 non-null  object 
 7   immig              103658 non-null  object 
 8   index              103658 non-null  int64  
 9   is_student         76893 non-null   float64
 10  lfsstat            103658 non-null  int64  
 11  naics_21           103658 non-null  object 
 12  prov               103658 non-null  int64  
 13  prov_name          103658 non-null  object 
 14  schooln            76893 non-null   float64
 15  sex                103658 non-null  object 
 16  st

### 
The dataframe has a total of 18 different columns. Those columns only use the following types: `[object, int64 and float64]`. We can determine that the boolean values, for instance, `is_student` has been encoded to a float when importing the data in mysql.

### Null and Non-null values
A few columns with 103658 non-null values have null values at all since there is a total of 103658 entries.

It is interesting to note that the new columns we have created in the cleaning process have no non-null values, since the mapping from the source column includes setting a non-null value even when the source value in `NaN.

For instance, if the original value for `cowmain` is `NaN`, the corresponding `cowmain_name` would be `'Not applicable'`

### Other observations
We also note that the source data has some columns such as `age_6` which is missing source information since we would expect any person to have an age. This would probably be attributable to the inability to acquire this data during the data collection phase, when creating the dataset. 